docs/BYTESTRING.md

```diff
@@ -0,0 +1,50 @@
+---
+title: The ByteString Situation
+---
+
+This document discusses the `GenUnchecked` instance for `ByteString`.
+
+
+## History
+
+Up to (but not including) version `0.4.0.0` of `genvalidity-bytestring`, the
+`GenUnchecked` instance of `ByteString` and `Text` were principled and would generate
+values that were significantly dangerous (to the point where they would cause segfaults).
+
+From version `0.4.0.0` of `genvalidity-bytestring`, these instances call `error` instead.
+The `GenInvalid` instance have been removed as well.
+
+## Considerations
+
+- Unchecked `ByteString`s are checkable for validity.
```

Comment:
`ByteString`s can't be fully checked, unless offset or length are not known to be 100% correct

Reply:
Good point!